### PR TITLE
[CT-629] Fix entryPrice calc

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
Fix entryPrice calc

### Test Plan
Unit tested

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced handling of liquidation orders, including improved calculations for entry prices and position sizes.
- **Bug Fixes**
	- Updated test cases to ensure accurate processing of order fills and liquidation events, including edge cases for validation failures.
- **Documentation**
	- Improved comments and documentation within functions for better clarity on parameters and return values.

### CODE REVIEW

File: indexer/services/ender/__tests__/handlers/order-fills/liquidation-handler.test.ts [hunk 1]
```diff
@@ -392,7 +392,7 @@ describe('LiquidationHandler', () => {
             defaultPerpetualPosition.openEventId,
           ),
           {
-            sumOpen: Big(defaultPerpetualPosition.size).plus(totalFilled).toFixed(),
+            sumOpen: Big(defaultPerpetualPosition.sumOpen!).plus(totalFilled).toFixed(),
             entryPrice: getWeightedAverage(
               defaultPerpetualPosition.entryPrice!,
               defaultPerpetualPosition.size,
```
Comment (dydxwill) [inline indexer/services/ender/__tests__/handlers/order-fills/liquidation-handler.test.ts:398]: not that entryPrice calc in unit test was already using prior size.

File: indexer/services/ender/__tests__/handlers/order-fills/order-handler.test.ts [hunk 2]
```diff
@@ -439,7 +439,7 @@ describe('OrderHandler', () => {
             defaultPerpetualPosition.openEventId,
           ),
           {
-            sumOpen: Big(defaultPerpetualPosition.size).plus(totalFilled).toFixed(),
+            sumOpen: Big(defaultPerpetualPosition.sumOpen!).plus(totalFilled).toFixed(),
             entryPrice: getWeightedAverage(
               defaultPerpetualPosition.entryPrice!,
               defaultPerpetualPosition.size,
```
Comment (dydxwill) [inline indexer/services/ender/__tests__/handlers/order-fills/order-handler.test.ts:445]: not that entryPrice calc in unit test was already using prior size.

**Review by vincentwschau — APPROVED** [review]

**Comment by dydxwill** [comment]
@mergifyio backport release/indexer/v6.x

**Comment by dydxwill** [comment]
@mergifyio backport release/indexer/v5.x
